dotCMS/src/main/java/com/dotcms/browser/BrowserAPI.java

```diff
@@ -0,0 +1,583 @@
+                        folderMap.put("permissions", permissions);
+                        folderMap.put("parent", folder.getInode());
+                        folderMap.put("mimeType", "");
+                        folderMap.put("name", folder.getName());
+                        folderMap.put("title", folder.getName());
+                        folderMap.put("description", folder.getTitle());
+                        folderMap.put("extension", "folder");
+                        folderMap.put("hasTitleImage", "");
+                        folderMap.put("__icon__", "folderIcon");
+                        returnList.add(folderMap);
+                    }
+                }
+            }
+        }
+
+        if (browserQuery.showLinks) {
+
+            // Getting the links directly under the parent folder or host
+            List<Link> links = new ArrayList<Link>();
+
+            if (parent != null) {
+                if (browserQuery.showWorking) {
+                    links.addAll(folderAPI.getLinks(parent, true, false, browserQuery.user, false));
+                } else {
+                    links.addAll(folderAPI.getLiveLinks(parent, browserQuery.user, false));
+                }
+                if (browserQuery.showArchived)
+                    links.addAll(folderAPI.getLinks(parent, true, browserQuery.showArchived, browserQuery.user, false));
+            } else {
+                links = folderAPI.getLinks(host, true, browserQuery.showArchived, browserQuery.user, false);
+            }
+
+
+            for (Link link : links) {
+
+                List<Integer> permissions2 = permissionAPI.getPermissionIdsFromRoles(link, roles, browserQuery.user);
+
+                if (permissions2.contains(PERMISSION_READ)) {
+                    Map<String, Object> linkMap = link.getMap();
+                    linkMap.put("permissions", permissions2);
+                    linkMap.put("mimeType", "application/dotlink");
+                    linkMap.put("name", link.getTitle());
+                    linkMap.put("title", link.getName());
+                    linkMap.put("description", link.getFriendlyName());
+                    linkMap.put("extension", "link");
+                    linkMap.put("hasLiveVersion", APILocator.getVersionableAPI().hasLiveVersion(link));
+                    linkMap.put("statusIcons", UtilHTML.getStatusIcons(link));
+                    linkMap.put("hasTitleImage", "");
+                    linkMap.put("__icon__", "linkIcon");
+                    returnList.add(linkMap);
+                }
+            }
+
+
+        }
+
+        StringBuilder luceneQuery = new StringBuilder();
+
+        List<String> baseTypes = new ArrayList<>();
+        if(browserQuery.showDotAssets) {
+            baseTypes.add("9");
+        }
+        if(browserQuery.showFiles) {
+            baseTypes.add("4");
+        }
+        if(browserQuery.showPages) {
+            baseTypes.add("5");
+        }
+        if(baseTypes.size()==0) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Substitute calls to size() == 0 (or size() != 0, size() > 0, size() < 1) with calls to isEmpty()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)